node/pkg/query/request.go

```diff
@@ -860,8 +918,140 @@ func (left *EthCallWithFinalityQueryRequest) Equal(right *EthCallWithFinalityQue
+//
+
+func (e *SolanaAccountQueryRequest) Type() ChainSpecificQueryType {
+	return SolanaAccountQueryRequestType
+}
+
+// Marshal serializes the binary representation of a Solana sol_account request.
+// This method calls Validate() and relies on it to range checks lengths, etc.
+func (saq *SolanaAccountQueryRequest) Marshal() ([]byte, error) {
+	if err := saq.Validate(); err != nil {
+		return nil, err
+	}
+
+	buf := new(bytes.Buffer)
+
+	vaa.MustWrite(buf, binary.BigEndian, uint32(len(saq.Commitment)))
+	buf.Write([]byte(saq.Commitment))
+
+	vaa.MustWrite(buf, binary.BigEndian, saq.MinContextSlot)
+	vaa.MustWrite(buf, binary.BigEndian, saq.DataSliceOffset)
+	vaa.MustWrite(buf, binary.BigEndian, saq.DataSliceLength)
+
+	vaa.MustWrite(buf, binary.BigEndian, uint8(len(saq.Accounts)))
+	for _, acct := range saq.Accounts {
+		buf.Write(acct[:])
+	}
+	return buf.Bytes(), nil
+}
+
+// Unmarshal deserializes a Solana sol_account query from a byte array
+func (saq *SolanaAccountQueryRequest) Unmarshal(data []byte) error {
+	reader := bytes.NewReader(data[:])
+	return saq.UnmarshalFromReader(reader)
+}
+
+// UnmarshalFromReader  deserializes a Solana sol_account query from a byte array
+func (saq *SolanaAccountQueryRequest) UnmarshalFromReader(reader *bytes.Reader) error {
+	len := uint32(0)
+	if err := binary.Read(reader, binary.BigEndian, &len); err != nil {
+		return fmt.Errorf("failed to read commitment len: %w", err)
+	}
+
+	if len > SolanaMaxCommitmentLength {
+		return fmt.Errorf("commitment string is too long, may not be more than %d characters", SolanaMaxCommitmentLength)
+	}
+
+	commitment := make([]byte, len)
+	if n, err := reader.Read(commitment[:]); err != nil || n != int(len) {
+		return fmt.Errorf("failed to read commitment [%d]: %w", n, err)
+	}
+	saq.Commitment = string(commitment)
+
+	if err := binary.Read(reader, binary.BigEndian, &saq.MinContextSlot); err != nil {
+		return fmt.Errorf("failed to read min slot: %w", err)
+	}
+
+	if err := binary.Read(reader, binary.BigEndian, &saq.DataSliceOffset); err != nil {
+		return fmt.Errorf("failed to read data slice offset: %w", err)
+	}
+
+	if err := binary.Read(reader, binary.BigEndian, &saq.DataSliceLength); err != nil {
+		return fmt.Errorf("failed to read data slice length: %w", err)
+	}
+
+	numAccounts := uint8(0)
+	if err := binary.Read(reader, binary.BigEndian, &numAccounts); err != nil {
+		return fmt.Errorf("failed to read number of account entries: %w", err)
+	}
+
+	for count := 0; count < int(numAccounts); count++ {
+		account := [SolanaPublicKeyLength]byte{}
+		if n, err := reader.Read(account[:]); err != nil || n != SolanaPublicKeyLength {
+			return fmt.Errorf("failed to read account [%d]: %w", n, err)
+		}
+		saq.Accounts = append(saq.Accounts, account)
+	}
+
+	return nil
+}
+
+// Validate does basic validation on a Solana sol_account query.
+func (saq *SolanaAccountQueryRequest) Validate() error {
+	if len(saq.Commitment) > math.MaxUint32 {
```

Comment:
Stupid question, but why can `len(saq.Commitment)` be > `SolanaMaxCommitmentLength`?

Reply:
Forgot to change this case once `SolanaMaxCommitmentLength` was added.

Fixed in PR #3728.